code/SQLite3Connector.php

```diff
@@ -48,7 +48,7 @@ public function getGeneratedID($table)
     public function getLastError()
     {
         $message = $this->dbConn->lastErrorMsg();
-        return $message === 'not an error' ? null : $message;
+        return $message === 'not an error' ? '' : $message;
```

Comment:
Instead of changing the value that is returned (which someone may be relying on being null and checking for that explicitly), it would be preferred to use `?? ''` in the place(s) the returned value is used that are actually causing the error (presumably this is the `preg_match()` you mention in the PR description)

Reply:
i reverted the changes

Reply:
Does the single change that remains resolve both problems in the PR description? Or is one of those not a problem?

Reply:
@GuySartorelli it seems that https://github.com/silverstripe/silverstripe-sqlite3/commit/f9dfa9f34c9c667db6522f32080debdbca0c0081 already fixed it but i didn't get that on my machine (i don't think it is released yet?)

Reply:
Makes sense. This isn't an officially supported module so it just gets releases on an ad-hoc basis. I'll make a release after this is merged.